vars/branchRepo.groovy

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/groovy
+
+def call(String gitRepoUrl, String gitRef, String checkoutDir, String branchName, Closure body) {
+    checkoutGitRepo {
+        repoUrl = gitRepoUrl
+        branch = gitRef
+        targetDir = checkoutDir
+    }
+    dir(checkoutDir) {
+        sshagent(['jenkinsgithub']) {
```

Comment:
@grdryn I'm going to leave this for now as i want to think about the best way to do this, we have the same problem in a few places so want to come up with a consistent solution.